src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.Unix.cs

```diff
@@ -61,5 +61,97 @@ internal static unsafe void GetAnsiStringBytes(ReadOnlySpan<char> chars, Span<by
             int actualByteLength = Encoding.UTF8.GetBytes(chars, bytes);
             bytes[actualByteLength] = 0;
         }
+
+        public static IntPtr AllocHGlobal(IntPtr cb)
+        {
+            nuint cbNative = (nuint)(nint)cb;
```

Comment:
Without the double cast, It would fail to compile with `error CS0030: Cannot convert type 'System.IntPtr' to 'nuint'` or `Cannot implicitly convert type 'nint' to 'nuint'. An explicit conversion exists (are you missing a cast?)`.

It has to do with the fact that `IntPtr` is weird type with unnaturally defined operations. The double casts are not worst. The worst is when the code does not do what you want without any compiler warnings, like what happened in https://github.com/dotnet/runtime/pull/41198